Zend/zend.c

```diff
@@ -1639,7 +1639,7 @@ ZEND_API ZEND_COLD ZEND_NORETURN void zend_error_noreturn(int type, const char *
 
 ZEND_API ZEND_COLD ZEND_NORETURN void zend_strerror_noreturn(int type, int errn, const char *message)
 {
-#ifdef HAVE_STR_ERROR_R
+#ifdef HAVE_STRERROR_R
 	char buf[1024];
 	strerror_r(errn, buf, sizeof(buf));
```

Comment:
@dstogov: I don't think this fully resolves the issue I mentioned in https://github.com/php/php-src/commit/067df263448ee26013cddee1065bc9c1f028bd23#commitcomment-121630134.

The GNU version of [`strerror_r(3)`](https://man7.org/linux/man-pages/man3/strerror.3.html) _may_ return a pointer to an immutable string instead of storing the result in the `buf` parameter. The XSI version always sets `buf` and returns an integer to indicate success/error.

Again, I don't think this would result in a build error. The `zend_error_noreturn()` line below would just end up printing `message` and `errn` with nothing between since `buf` would remain empty.

Reply:
> since buf would remain empty

Correction: `buf` would be uninitialized. I was incorrectly thinking of of static/global vars when I wrote that, which are zero-allocated.